### ISSUE TITLE
RawTransactionInfo(Transaction) Always Fails

### ISSUE THREAD

**Issue by pkgmain** [ConsensusJ/consensusj#35]
`RawTransactionInfo`'s constructor that takes a `Transaction` loops over the `TransactionInput`s to populate the `VinList`. When instantiating the `Vin`, it uses `input.getHash()` as an argument. However, `TransactionInput` has not overridden `getHash()` and therefore throws an `UnsupportedOperationException` from `Message`. It's unclear to me how this ever works. I'd put in a PR, but if I'm being honest, I don't know enough about Bitcoin to know what should be happening here.

RawTransactionInfo.java
```java
        vin = new VinList();
        for (TransactionInput input : transaction.getInputs()) {
            vin.add(new Vin(input.getHash(),
                            input.getOutpoint().getIndex(),
                            input.getScriptSig().toString(),
                            input.getSequenceNumber()));
        }
```

Message.java
```java
    public Sha256Hash getHash() {
        throw new UnsupportedOperationException();
    }
```

![screen shot 2018-05-10 at 4 12 15 pm](https://user-images.githubusercontent.com/5573133/39891952-0c0a3658-546d-11e8-8e36-aa15d1ec6c5c.png)

**Comment by msgilligan** [comment]
I think you found a bug. It looks like the `RawTransactionInfo` constructor that takes a `Transaction` has never been used or tested before.

I think the correct code should be:
```
        vin = new VinList();
        for (TransactionInput input : transaction.getInputs()) {
            vin.add(new Vin(txid,
                            input.getOutpoint().getIndex(),
                            input.getScriptSig().toString(),
                            input.getSequenceNumber()));
        }
```

**Comment by msgilligan** [comment]
https://github.com/ConsensusJ/consensusj/pull/36 should fix the problem. @barrongineer can you review the PR and/or test the change in your code?

**Comment by msgilligan** [comment]
Closing as fixed (#36)
